### ISSUE TITLE
Fail to build 0.41.0

### ISSUE THREAD

**Issue by 0323pin** [nushell/nushell#4191]
### Describe the bug

This is actually not a `nushell` issue, rather a `sysinfo` one but, just leaving a note here as well.

I've been maintaning `nushell` for NetBSD and using it myself from 0.34.0 through to 0.40.0 but, now I can not build 0.41.0 since, `sysinfo` fails to build which is rather unfortunate.

The error goes as follows,

```
error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:11:6
   |
11 | impl fmt::Display for ProcessStatus {
   |      ^^^ use of undeclared crate or module `fmt`

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:12:27
   |
12 |     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
   |                           ^^^ use of undeclared crate or module `fmt`

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:12:46
   |
12 |     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
   |                                              ^^^ use of undeclared crate or module `fmt`

error[E0412]: cannot find type `ProcessStatus` in this scope
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:11:23
   |
11 | impl fmt::Display for ProcessStatus {
   |                       ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
7  | use crate::ProcessStatus;
   |

error[E0412]: cannot find type `ProcessStatus` in this scope
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:73:25
   |
73 |     fn status(&self) -> ProcessStatus {
   |                         ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
7  | use crate::ProcessStatus;
   |

error[E0433]: failed to resolve: use of undeclared type `ProcessStatus`
  --> /usr/pkgsrc/wip/nushell/work/vendor/sysinfo-0.21.1/src/unknown/process.rs:74:9
   |
74 |         ProcessStatus::Unknown(0)
   |         ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
7  | use crate::ProcessStatus;
   |

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `sysinfo` due to 6 previous errors
```

I'm sure I could patch `nushell` to use `sysinfo`-0.20.2 which it's the version required by `nushell`-0.40.0 but, this would be a very short lived fix.

I'll report this on `sysinfo` crate and hope that I can keep using `nushell` as my default shell going forward.

### How to reproduce

Build 0.41.0 on NetBSD with Rust-1.56.1

### Expected behavior

NA

### Screenshots

NA

### Configuration

/home/pin()
2021-12-08 10:50 > version | pivot key value | to md
|key|value|
|-|-|
|version|0.40.0|
|branch|master|
|short_commit|39153dd1|
|commit_hash|39153dd17006ffb94ab22bf5c7c5f133155a8d6d|
|commit_date|2021-11-16 01:12:23 +00:00|
|build_os|netbsd-x86_64|
|rust_version|rustc 1.56.0|
|cargo_version|cargo 1.56.0|
|pkg_version|0.40.0|
|build_time|2021-11-17 11:04:16 +01:00|
|build_rust_channel|release|
|features|ctrlc, dataframe, default, rustyline, term, which, zip|
|installed_plugins|fetch, inc, match, post, ps, sys, textview|

### Additional context

NA

**Comment by kubouch** [comment]
I'm not sure if there is anything Nushell can do. I guess we could accept a PR that downgrades it to the last working version, as long as it doesn't break anything. `sysinfo` seems to be required only for `sys` and `ps`, so another workaround would be to build without these features. But obviously, that is not a good solution for package distribution.

**Comment by 0323pin** [comment]
> I'm not sure if there is anything Nushell can do. I guess we could accept a PR that downgrades it to the last working version, as long as it doesn't break anything. `sysinfo` seems to be required only for `sys` and `ps`, so another workaround would be to build without these features. But obviously, that is not a good solution for package distribution.

I'm aware of this. I got some feedback from `sysinfo` and the work needs to be done there. Or, apparently it needs to be done in `libc` itself 😞 

I don't think `nushell` needs or, should do anything. I've just posted FYI.
I can apply a downgrade patch myself but, I don't want this to be permanent, it will cause issues sooner or later.

**Comment by 0323pin** [comment]
I've patched `nushell` to use `sysinfo`-0.20.4 for now and will merge the package update with this patch later today.
`sysinfo` has a fix release in the works as well, so hopefully this won't happen again.

|key|value|
|-|-|
|version|0.41.0|
|branch|master|
|short_commit|2d330f5c|
|commit_hash|2d330f5c77275014b24b1624045a77a52ec6bf57|
|commit_date|2021-12-06 21:36:34 +00:00|
|build_os|netbsd-x86_64|
|rust_version|rustc 1.56.1|
|cargo_version|cargo 1.56.0|
|pkg_version|0.41.0|
|build_time|2021-12-08 14:41:44 +01:00|
|build_rust_channel|release|
|features|ctrlc, dataframe, default, rustyline, term, which, zip|
|installed_plugins|fetch, inc, match, post, ps, sys, textview|


Closing this issue now.
